File: Data Structures & Algorithms/max-area-of-island/submission-1.py
class Solution:
    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
        res = 0
        N, M = len(grid), len(grid[0])

        def dfs(row, col):
            if (
                row < 0
                or row >= N
                or col < 0
                or col >= M
                or grid[row][col] == 0
            ):
                return 0
            grid[row][col] = 0
            return 1 + dfs(row + 1, col) + dfs(row, col + 1) + dfs(row - 1, col) + dfs(row, col - 1)

        for row in range(N):
            for col in range(M):
                if grid[row][col] == 1:
                    res = max(res, dfs(row, col))
        return res

File: Data Structures & Algorithms/valid-sudoku/submission-1.java
public class Solution {
    
    public boolean isValidSudoku(char[][] board) {
        boolean[][] row = new boolean[9][9];
        boolean[][] col = new boolean[9][9];
        boolean[][] block = new boolean[9][9];
        for (int r = 0; r < 9; r++) {
            for (int c = 0; c < 9; c++) {
                if (board[r][c] == '.') {
                    continue;
                }
                int val = board[r][c] - '1';
                if (row[r][val]) {
                    return false;
                }
                row[r][val] = true;
                if (col[c][val]) {
                    return false;
                }
                col[c][val] = true;
                int block_index = (r / 3) * 3 + (c / 3);
                if (block[block_index][val]) {
                    return false;
                }
                block[block_index][val] = true;
            }
        }
        return true;
    }
}
